pkg/process/checks/process.go

```diff
@@ -40,11 +41,12 @@ const (
 )
 
 // NewProcessCheck returns an instance of the ProcessCheck.
-func NewProcessCheck(config ddconfig.Reader) *ProcessCheck {
+func NewProcessCheck(config ddconfig.Reader, sysprobeYamlConfig ddconfig.Reader) *ProcessCheck {
 	check := &ProcessCheck{
-		config:        config,
-		scrubber:      procutil.NewDefaultDataScrubber(),
-		lookupIdProbe: NewLookupIDProbe(config),
+		config:             config,
+		scrubber:           procutil.NewDefaultDataScrubber(),
+		lookupIdProbe:      NewLookupIDProbe(config),
+		sysprobeYamlConfig: sysprobeYamlConfig,
```

Comment:
Optional: Personally, I'd just create a new service extractor using the config here, and set the `serviceExtractor` field, instead of saving the sysprobe config. It's not used for anything else.

Reply:
this makes sense to me too